apis/infrastructure/v1beta1/k8sinstallerconfig_types.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// K8sInstallerConfigSpec defines the desired state of K8sInstallerConfig
+type K8sInstallerConfigSpec struct {
+	// BundleRepo is the registry from which bundle will be downloaded
+	BundleRepo string `json:"bundleRepo"`
+
+	// BundleType is the type of bundle (e.g. k8s, tinyk8s) need to be downloaded
+	BundleType string `json:"bundleType"`
+}
+
+// K8sInstallerConfigStatus defines the observed state of K8sInstallerConfig
+type K8sInstallerConfigStatus struct {
+	// Ready indicates the InstallationSecret field is ready to be consumed
+	// +optional
+	Ready bool `json:"ready,omitempty"`
+
+	// InstallationSecret is an optional reference to a generated installation secret by K8sInstallerConfig controller
```

Comment:
```suggestion
	// InstallationSecret is an optional reference to a generated installation secret by K8sInstaller controller
```

Reply:
@dharmjit , isn't the name of the controller watching K8sInstallerConfig CR will be K8sInstallerConfig? or we going to create K8sInstaller controller watching K8sInstallerConfig CR?

Reply:
Ohh, my bad I guess it makes sense to keep the CRD and Controller name aligned.